cmd/runtimecfg/node-ip.go

```diff
@@ -138,33 +138,52 @@ func set(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func checkAddressUsable(chosen []net.IP) (err error) {
+	// If using IPv6, verify that the choosen address isn't tentative
+	// i.e. we can actually bind to it
+	if len(chosen) > 0 && net.IPv6len == len(chosen[0]) {
+		_, err = net.Listen("tcp", "["+chosen[0].String()+"]:")
+		if err != nil {
+			log.Errorf("Chosen node IP is not usable")
+			return err
+		}
+	}
+	return err
+}
+
 func getSuitableIPs(retry bool, vips []net.IP, preferIPv6 bool) (chosen []net.IP, err error) {
 	// Enable debug logging in utils package
 	utils.SetDebugLogLevel()
 	for {
 		if len(vips) > 0 {
 			chosen, err = utils.AddressesRouting(vips, utils.ValidNodeAddress)
 			if len(chosen) > 0 || err != nil {
-
-				// If using IPv6, verify that the choosen address isn't tentative
-				// i.e. we can actually bind to it
-				if len(chosen) > 0 && net.IPv6len == len(chosen[0]) {
-					_, err := net.Listen("tcp", "["+chosen[0].String()+"]:")
-					if err != nil {
-						log.Errorf("Chosen node IP is not usable")
-						if !retry {
-							return nil, fmt.Errorf("Failed to find node IP")
-						}
-						time.Sleep(time.Second)
-						continue
+				if err == nil {
+					err = checkAddressUsable(chosen)
+				}
+				if err != nil {
```

Comment:
I am wondering if this logic should go to below line 147 and the line 161 inside a if like: if len(choosen) > 0 && err == nil {

Reply:
Initially I was going to do this, but because there's a loop flow control statement I don't think we can put this block in the function.